Lib/idlelib/ParenClose.py

```diff
@@ -0,0 +1,91 @@
+"""
+Parens and Ticks Closing Extension
+
+When you hit left paren or tick,
+automatically creates the closing paren or tick.
+
+Author: Charles M. Wohlganger
+        charles.wohlganger@gmail.com
+
+Last Updated: 12-Aug-2017 by Charles Wohlganger
+
+Add to the end of config-extensions.def :
+
+    [ParenClose]
+    enable = True
+    paren_close = True
+    tick_close = True
+    skip_closures = True
+    comment_space = True
+    [ParenClose_cfgBindings]
+    p-open = <Key-parenleft> <Key-bracketleft> <Key-braceleft>
+    t-open = <Key-'> <Key-">
+    p-close = <Key-parenright> <Key-bracketright> <Key-braceright>
```

Comment:
p-open, t-open, and p-close appear in the keybindings list in Configure IDLE.  Was the intention to reserve these so the keys couldn't be bound to another shortcut?  If so, I think they need to be suppressed from the keybindings selection, but still seen as a duplicate if someone tries to select it.

Reply:
If this were implemented as an extension, which it will not be, the section title should be [ParenClose_bindings] because these are fixed, not configurable, bindings.  The user reconfiguration dialog should not allow bare, unmodified <Key-x> bindings, at least not for character keys.